File: app/src/main/java/de/christinecoenen/code/zapp/app/mediathek/api/request/Query.java
package de.christinecoenen.code.zapp.app.mediathek.api.request;


import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Query implements Serializable {

	private final List<String> fields = new ArrayList<>();
	private final String query;

	Query(String queryString, String... fieldNames) {
		Collections.addAll(fields, fieldNames);
		query = queryString;
	}

	@Override
	public String toString() {
		return "Query{" +
			"fields=" + fields +
			", query='" + query + '\'' +
			'}';
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/app/mediathek/api/request/Query.kt
package de.christinecoenen.code.zapp.app.mediathek.api.request

import androidx.annotation.Keep
import java.io.Serializable

@Keep
internal data class Query(val query: String, val fields: List<String>) : Serializable {

	constructor(query: String, vararg fields: String) : this(query, fields.toList())

}
